### PR TITLE
Move acset homomorphism search into own submodule

### PR DESCRIPTION
**Pull request by kris-brown**
This PR simply migrates some code from `CSets.jl` to a new file `HomSearch.jl`. Addresses https://github.com/AlgebraicJulia/Catlab.jl/issues/774

(Also a minor cleanup of replacing `out_homs(S,c)` and `in_homs(S,c)` with `homs(S; to=c)` and `homs(S; from=c)`.)

### CODE REVIEW

File: src/categorical_algebra/HomSearch.jl [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+using DataStructures: BinaryHeap, DefaultDict
+
+
+"""
+Functionality related to search problems involving ACSets, e.g.:
+
+- enumerating Hom(X,Y) where X,Y are ACSets
+- enumerating subobjects of an ACSet, X
+- enumerating partial overlaps between ACSets
+"""
```
Comment (epatters) [inline src/categorical_algebra/HomSearch.jl:25]: This docstring can become the module-level docstring, i.e., be moved to the very top of the file.

**Review by epatters — CHANGES_REQUESTED** [review]
Thanks Kris! This looks good apart from one small comment.

**Review by epatters — CHANGES_REQUESTED** [review]
Sorry, I didn't notice this the first time, but shouldn't we also reorganize the tests accordingly?

**Comment by kris-brown** [comment]
Good point! There are still some calls to `homomorphism` in `CSets.jl` but the dedicated hom search and subobject search is now moved

**Review by epatters — APPROVED** [review]
Thanks Kris!

**Pull request merged**